core/Archive/ArchiveInvalidator.php

```diff
@@ -528,20 +535,48 @@ public function removeInvalidations($idSite, $plugin, $report = null)
     }
 
     /**
-     * Re-archives reports without propagating exceptions.
+     * Schedules a re-archiving reports without propagating exceptions. This is scheduled
+     * since adding invalidations can take a long time and delay UI response times.
      *
      * @param int|int[]|'all' $idSites
      * @param string $pluginName
      * @param string|null $report
      * @param Date|null $startDate
      */
-    public function reArchiveReportSafely($idSites, string $pluginName, string $report = null, Date $startDate = null)
+    public function scheduleReArchiving($idSites, string $pluginName, string $report = null, Date $startDate = null)
     {
         try {
-            $this->reArchiveReport($idSites, $pluginName, $report, $startDate);
+            $reArchiveList = new ReArchiveList($this->logger);
+            $reArchiveList->add([$idSites, $pluginName, $report, $startDate ? $startDate->getTimestamp() : null]);
         } catch (\Throwable $ex) {
-            $logger = StaticContainer::get(LoggerInterface::class);
-            $logger->info("Failed to schedule rearchiving of past reports for $pluginName plugin.");
+            $this->logger->info("Failed to schedule rearchiving of past reports for $pluginName plugin.");
+        }
+    }
+
+    /**
+     * Applies the queued archiving rearchiving entries.
+     */
+    public function applyScheduledReArchiving()
+    {
+        $reArchiveList = new ReArchiveList($this->logger);
+        $items = $reArchiveList->getAll();
+
+        foreach ($items as $entry) {
+            try {
+                list($idSites, $pluginName, $report, $startDateTs) = $entry;
+
+                $this->reArchiveReport($idSites, $pluginName, $report, Date::factory($startDateTs));
+            } catch (\Throwable $ex) {
+                $this->logger->info("Failed to create invalidations for report re-archiving (idSites = {idSites}, pluginName = {pluginName}, report = {report}, startDate = {startDateTs}): {ex}", [
+                    'idSites' => json_encode($idSites),
+                    'pluginName' => $pluginName,
+                    'report' => $report,
+                    'startDateTs' => $startDateTs,
+                    'ex' => $ex,
+                ]);
+            } finally {
+                $reArchiveList->remove([$entry]);
```

Comment:
Does it make sense to remove the record even if it failed to create the invalidation? Shouldn't we keep it and try to add it another time?

Reply:
No preference from my side here as I reckon usually it shouldn't fail? if there may be some reasons why it might happen could also keep the entry and retry later

Reply:
It should only fail if the input is broken for some reason, this is mostly to replicate the old behavior where if it failed for some reason, invalidations wouldn't be added.

Reply:
Now that the entries are json encoded, that needs to be done for the removal as well. Otherwise it won't be removed at all. Alternatively you could also use another variable for the decoded entry...

```suggestion
                $reArchiveList->remove([json_encode($entry)]);
```